File: modules/expo-template-native/ios/ExpoTemplateNativeModule.swift
import ExpoModulesCore
import UIKit

public class ExpoTemplateNativeModule: Module {
  public func definition() -> ModuleDefinition {
    Name("ExpoTemplateNative")

    Function("getRuntimeInfo") {
      return [
        "platform": "ios",
        "systemVersion": UIDevice.current.systemVersion,
      ]
    }
  }
}

File: modules/expo-template-native/android/src/main/java/expo/modules/expotemplatenative/ExpoTemplateNativeModule.kt
package expo.modules.expotemplatenative

import android.os.Build
import expo.modules.kotlin.modules.Module
import expo.modules.kotlin.modules.ModuleDefinition

class ExpoTemplateNativeModule : Module() {
  override fun definition() = ModuleDefinition {
    Name("ExpoTemplateNative")

    Function("getRuntimeInfo") {
      mapOf(
        "platform" to "android",
        "systemVersion" to Build.VERSION.RELEASE,
      )
    }
  }
}
